ultra/ultra/evaluate.py

```diff
@@ -76,37 +77,83 @@ def evaluation_check(
         )
         agents[agent_id].save(checkpoint_directory)
 
-        # Perform the evaluation on this agent and save the data.
-        evaluation_data.update(
-            ray.get(
-                [
-                    evaluate.remote(
-                        seed=episode.eval_count,
-                        experiment_dir=episode.experiment_dir,
-                        agent_ids=[agent_id],
-                        policy_classes={agent_id: policy_classes[agent_id]},
-                        checkpoint_dirs={agent_id: checkpoint_directory},
-                        scenario_info=scenario_info,
-                        num_episodes=eval_episodes,
-                        max_episode_steps=max_episode_steps,
-                        headless=headless,
-                        timestep_sec=timestep_sec,
-                        log_dir=log_dir,
-                    )
-                ]
-            )[0]
+        evaluation_task_id = evaluate.remote(
+            seed=episode.eval_count,
+            experiment_dir=episode.experiment_dir,
+            agent_ids=[agent_id],
+            policy_classes={agent_id: policy_classes[agent_id]},
+            checkpoint_dirs={agent_id: checkpoint_directory},
+            scenario_info=scenario_info,
+            num_episodes=eval_episodes,
+            max_episode_steps=max_episode_steps,
+            headless=headless,
+            timestep_sec=timestep_sec,
+            log_dir=log_dir,
+            eval_mode=False,
+        )
+        evaluation_task_ids[evaluation_task_id] = (
+            episode.get_itr(agent_id),
+            episode,
+            "eval_train",
+        )
+
+    for agent_id in agent_ids_to_evaluate:
```

Comment:
There were problems before when we had these `evalute.remote(..)` calls in the same `for agent_id in agent_ids_to_evaluate: ...` loop, correct? I wonder if it works now since the evaluations happen asynchronously? I can test this too.

Reply:
I am not 100% sure which issue youre referring to but it would be good to test!

Reply:
Sorry yeah, see #744 for the changes I was thinking of. Let me know your thoughts. I didn't run into any problems with those changes either.